lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -131,7 +133,8 @@ def _get_user_partition(self, user_partition_id):
         Returns the user partition with the specified id.  Raises
         `NoSuchUserPartitionError` if the lookup fails.
         """
-        for user_partition in self.user_partitions:
+        partition_service = self.runtime.service(self, 'partitions')
```

Comment:
Defaulting to self.user_partitions should fix all my failing unit tests, but I don't feel great about it. Though if I don't provide this fallback, I should change the decorator to _XBlock.needs_.

@cpennington @nasthagiri thoughts?